corehq/apps/userreports/pillow.py

```diff
@@ -66,6 +67,16 @@ def _filter_by_hash(configs, ucr_division):
     return filtered_configs
 
 
+def _filter_missing_domains(configs):
+    """Return a list of configs whose domain exists on this environment"""
+    domain_names = [config.domain for config in configs if config.is_static]
+    existing_domains = list(get_domain_ids_by_names(domain_names))
+    return [
+        config for config in configs
+        if not config.is_static or config.domain in existing_domains
+    ]
```

Comment:
The following feels a little simpler to me and I think it should be the same behavior. Thoughts on this?

```python
domain_names = {config.domain for config in configs if config.is_static}
existing_domains = set(get_domain_ids_by_names(domain_names).keys())
valid_domains = existing_domains.intersection(domain_names)
return [config for config in configs if config.domain in valid_domains)
```